centreon/www/front_src/src/ResourceAccessManagement/Listing/useListing.ts

```diff
@@ -0,0 +1,136 @@
+import { useEffect, useState } from 'react';
+
+import { prop } from 'ramda';
+import { useAtom, useAtomValue, useSetAtom } from 'jotai';
+
+import { useFetchQuery } from '@centreon/ui';
+
+import {
+  resourceAccessManagementSearchAtom,
+  selectedRowsAtom,
+  resourceAccessRulesNamesAtom
+} from '../atom';
+import {
+  Listing,
+  ResourceAccessRuleType,
+  ResourceAccessRuleListingType
+} from '../models';
+import type { SortOrder } from '../models';
+
+import { listingDecoder } from './api/decoders';
+import { buildResourceAccessRulesEndpoint } from './api/endpoints';
+import useListingColumns from './columns';
+
+const useListing = (): Listing => {
+  const columns = useListingColumns();
+  const [selectedColumnIds, setSelectedColumnIds] = useState(
+    columns.map(prop('id'))
+  );
+  const resetColumns = (): void => {
+    setSelectedColumnIds(columns.map(prop('id')));
+  };
+
+  const [limit, setLimit] = useState<number | undefined>(10);
+  const [page, setPage] = useState<number | undefined>(undefined);
+  const [sortF, setSortF] = useState<string>('name');
+  const [sortO, setSortO] = useState<SortOrder>('asc');
+  const [selectedRows, setSelectedRows] = useAtom(selectedRowsAtom);
+  const searchValue = useAtomValue(resourceAccessManagementSearchAtom);
+  const setResourceAccessRulesNames = useSetAtom(resourceAccessRulesNamesAtom);
```

Comment:
you can remove the atom  `resourceAccessRulesNamesAtom` if it is not being utilized